### ISSUE TITLE
Some node-exporter graphs have no data due to snap confinement - AppArmor denials

### ISSUE THREAD

**Issue by nobuto-m** [canonical/grafana-agent-operator#23]
### Bug Description

![image](https://github.com/canonical/grafana-agent-operator/assets/4356209/ceaabdbf-a5da-432b-bf4f-8e742db95635)

![image](https://github.com/canonical/grafana-agent-operator/assets/4356209/549eaf40-c289-4ceb-8ac2-2b1e0fad3043)

![image](https://github.com/canonical/grafana-agent-operator/assets/4356209/0781064d-787b-4f52-8d9e-11f299d286c1)

![image](https://github.com/canonical/grafana-agent-operator/assets/4356209/a546a513-b065-4f23-a93b-852ec70207ab)

### To Reproduce

With MAAS provider, deploy the grafana-agent charm onto another charm, and relate it to COS.

### Environment

```
$ snap info grafana-agent
...
services:
  grafana-agent: simple, enabled, active
snap-id:      T1u01xKcraA4IpWHpIGNU2nAOom9qG7i
tracking:     latest/edge
refresh-date: today at 11:55 UTC
channels:
  latest/stable:    0.34.3 2023-07-19 (13) 264MB -
  latest/candidate: 0.34.3 2023-07-19 (13) 264MB -
  latest/beta:      0.34.3 2023-07-19 (13) 264MB -
  latest/edge:      0.35.4 2023-08-15 (16) 282MB -
installed:          0.35.4            (16) 282MB -
```

### Relevant log output

```shell
Oct 26 12:11:57 up-snail kernel: audit: type=1400 audit(1698322317.541:1674): apparmor="DENIED" operation="open" profile="snap.grafana-agent.grafana-agent" name="/proc/spl/kstat/zfs/" pid=104040 comm="agent" requested_mask="r" denied_mask="r" fsuid=0 ouid=0
Oct 26 12:11:57 up-snail kernel: audit: type=1400 audit(1698322317.541:1675): apparmor="DENIED" operation="open" profile="snap.grafana-agent.grafana-agent" name="/sys/fs/btrfs/" pid=104040 comm="agent" requested_mask="r" denied_mask="r" fsuid=0 ouid=0
Oct 26 12:11:57 up-snail kernel: audit: type=1107 audit(1698322317.545:1676): pid=1053 uid=102 auid=4294967295 ses=4294967295 subj=unconfined msg='apparmor="DENIED" operation="dbus_method_call"  bus="system" path="/org/freedesktop/login1" interface="org.freedesktop.login1.Manager" member="ListSeats" mask="send" name="org.freedesktop.login1" pid=104040 label="snap.grafana-agent.grafana-agent" peer_pid=1064 peer_label="unconfined"
Oct 26 12:11:57 up-snail kernel: audit: type=1107 audit(1698322317.549:1677): pid=1053 uid=102 auid=4294967295 ses=4294967295 subj=unconfined msg='apparmor="DENIED" operation="dbus_method_call"  bus="system" path="/org/freedesktop/systemd1" interface="org.freedesktop.DBus.Properties" member="Get" mask="send" name="org.freedesktop.systemd1" pid=104040 label="snap.grafana-agent.grafana-agent" peer_pid=1 peer_label="unconfined"
Oct 26 12:11:57 up-snail kernel: audit: type=1107 audit(1698322317.549:1678): pid=1053 uid=102 auid=4294967295 ses=4294967295 subj=unconfined msg='apparmor="DENIED" operation="dbus_method_call"  bus="system" path="/org/freedesktop/systemd1" interface="org.freedesktop.systemd1.Manager" member="ListUnits" mask="send" name="org.freedesktop.systemd1" pid=104040 label="snap.grafana-agent.grafana-agent" peer_pid=1 peer_label="unconfined"
Oct 26 12:11:57 up-snail kernel: audit: type=1400 audit(1698322317.593:1679): apparmor="DENIED" operation="open" profile="snap.grafana-agent.grafana-agent" name="/proc/sys/kernel/threads-max" pid=104040 comm="agent" requested_mask="r" denied_mask="r" fsuid=0 ouid=0
```

**Comment by nobuto-m** [comment]
By running the binary outside of the confined context (calling `/snap/grafana-agent/16/agent -config.expand-env -config.file /etc/grafana-agent.yaml` on the host without snapd), I got the following working just for testing:

- Processes State
- PIDs Number and Limit

![image](https://github.com/canonical/grafana-agent-operator/assets/4356209/ceaaca89-9dab-4758-9c03-1fe37cfa71e3)

- Systemd Sockets
- Systemd Units States

![image](https://github.com/canonical/grafana-agent-operator/assets/4356209/4e8ce284-bb1d-4282-ab07-09cec634417c)

**Comment by cbartz** [comment]
We have the same problem. The logs are full of permission denied errors: https://pastebin.canonical.com/p/tg6vNpDP6K/

**Comment by cbartz** [comment]
A user mentioned ( https://chat.charmhub.io/charmhub/pl/mwhz1e1c37rdpbbfbzd19bqi3c ) that reverting to `stable/latest` fixed the issue. @sed-i @Abuelodelanada any ideas ?

**Comment by phvalguima** [comment]
Hi @cbartz, I am still seeing these `DENIED` in both cases. However, the issue I had (i.e. missing info that I needed for my use case) is not happening in latest/stable. I don't think the info that was missing, in my specific case, was due to the DENIEDs thou

**Comment by sed-i** [comment]
Will likely be fixed with #52.
